src/lib/ptree/json.lua

```diff
@@ -20,13 +35,20 @@ local function skip_whitespace(input)
    drop_while(input, whitespace_pat)
 end
 
-local function check(input, ch)
+local function peek(input, ch)
    if input:peek_char() ~= ch then return false end
-   input:read_char()
    return true
 end
 
-local function consume(input, expected)
+local function check(input, ch)
+   local v = peek(input, ch)
+   if v then
+      input:read_char()
+   end
+   return v
+end
```

Comment:
This `check` returns `v`, however all the uses of `check` are evaluated in a boolean context. I think the former implementation would work and it's not necessary to change it.

Reply:
Rather than revert to the exact former implementation, I've changed this to use the local `peek` function as the condition is already implemented in there. Shout out if you see any issue with that...